ICG produces uncompilable code for inner classes of class templates with scoped template arguments

```c++
namespace bar {
class Potato{};
}


class Monkey {
    public:
    class Ape {}; 
};

template<class T>
class Foo {
    public:
    class InnerFoo {}; 
    InnerFoo innerFoo;
};
```
```c++
class Sandbox : public Trick::SimObject {

    public:
    Foo<bar::Potato> foo;
    Foo<Monkey::Ape> foo2;

};
```

```
build/class_map.cpp:99:24: error: ‘attrFoo_classbar’ has not been declared
     extern ATTRIBUTES  attrFoo_classbar::Potato___InnerFoo[] ;
build/class_map.cpp:108:24: error: ‘attrFoo_classMonkey’ has not been declared
     extern ATTRIBUTES  attrFoo_classMonkey::Ape___InnerFoo[] ;
```

[`ConstructValues::mangle_template_param`](https://github.com/nasa/trick/blob/fd9551bb7951a856d5ad27efd4f1b5921b7f4f49/trick_source/codegen/Interface_Code_Gen/ConstructValues.cpp#L68) needs to additionally replace all instances of `:`

--- a/trick_source/codegen/Interface_Code_Gen/ConstructValues.cpp
+++ b/trick_source/codegen/Interface_Code_Gen/ConstructValues.cpp
@@ -70,6 +70,7 @@ static void mangle_template_param(std::string &work_string, const std::string & 
         std::replace( work_string.begin(), work_string.end(), '*', '_') ;
         std::replace( work_string.begin(), work_string.end(), '[', '_') ;
         std::replace( work_string.begin(), work_string.end(), ']', '_') ;
+        std::replace( work_string.begin(), work_string.end(), ':', '_') ;
         work_string.erase(std::remove_if(work_string.begin(), work_string.end(),
         (int(*)(int))std::isspace), work_string.end()) ;
     }

